### PR TITLE
More cmake improvements & fixes

### PR DESCRIPTION
**Pull request by Krzmbrzl**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,13 +16,13 @@ project(libmumble
...
 option(LIBMUMBLE_BUILD_TESTS "Build tests" ON)
-
 option(LIBMUMBLE_BUNDLED_GSL "Use the bundled GSL version instead of looking for one on the system" ON)
-
 option(LIBMUMBLE_STATIC "Build the library as static instead of shared" OFF)
+option(LIBMUMBLE_WARNINGS_AS_ERRORS "Treat warnings as errors" OFF)
 
 
 include(setup_dependencies)
+include(CompilerUtilities)
```
Comment (davidebeatrici) [inline CMakeLists.txt:25]: Why the naming inconsistency?

Comment (Krzmbrzl) [reply CMakeLists.txt:25]: Fair point. I'll change that.

**Comment by davidebeatrici** [comment]
I just noticed the position of `<cassert>` appears to be random, another `clang-format` quirk?

**Comment by Krzmbrzl** [comment]
No - this PR is apparently not yet fully clang-formatted. I introduce that with #11. From then on, the CI will check the formatting and it should remain consistent. However, running the formatting script on this PR would have created merge conflicts in #11, so I didn't do it.

**Pull request merged**